celements-model/src/main/java/com/celements/model/object/restriction/ObjectQueryBuilder.java

```diff
@@ -51,7 +54,22 @@ public ObjectQueryBuilder() {
    * NOTE: value may not be null, instead use {@link #filterAbsent(ClassField)}
    */
   public final @NotNull <T> B filter(@NotNull ClassField<T> field, @NotNull T value) {
-    return filter(new FieldRestriction<>(getBridge(), field, value));
+    if (isSingleSelectListField(field)) {
+      return filter(newListFieldRestriction(field, value));
+    } else {
+      return filter(new FieldRestriction<>(getBridge(), field, value));
+    }
+  }
+
+  private boolean isSingleSelectListField(ClassField<?> field) {
+    return (field instanceof ListField) && !((ListField<?>) field).isMultiSelect();
+  }
+
+  @SuppressWarnings("unchecked")

```

Comment:
add javadoc explaining which warnings are suppressed and why it is safe to ignore them.